### PR TITLE
PYTHON-2520 Support starting clusters with requireApiVersion

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
This is blocked for now because the tests hang in PyMongo when using a replicat_set.  It works for standalone and sharded clusters.

**Comment by blink1073** [comment]
Build: https://spruce.mongodb.com/version/6724ba5678c209000755f4ab/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by blink1073** [comment]
Here's a build without async/sll, and with pytest-timeout: https://spruce.mongodb.com/version/672927537f4cb000073704fe/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by blink1073** [comment]
I isolated the cause in the unit tests here: we can't write to a replica_set if either the primary or all members have requireApiversion set and the write concern is greater than 1.  It times out.

**Comment by ShaneHarvey** [comment]
Nice find, that sounds like a clear server bug although I'm not sure if they will care to prioritize fixing it since it's a test only feature. Still, could you open a SERVER bug ticket?

**Comment by blink1073** [comment]
Okay, it is a bit different than I thought.  Based on a prompt from @alcaeus I tried running it with `require-api-version.js` instead.  That lead me to adding a delay in startup until all members are reachable before setting requireApiVersion.  If you do that, then write operations succeed.  However, the tests still hung on `test/test_collection.py::TestCollection::test_create_index`.   So I tried running just `client.test.test.create_index("hello")` and that times out.  So at the very least that operation doesn't support it, but we don't know which others might be affected.  Do we still think it is worth raising a SERVER ticket?

**Comment by ShaneHarvey** [comment]
Yes it's still worth opening a bug ticket.

**Comment by blink1073** [comment]
https://jira.mongodb.org/browse/SERVER-97010

**Comment by blink1073** [comment]
Hmm, in the latest run of pymongo I'm still seeing unexpected behavior.  The same two tests are timing out when trying to drop a collection on a sharded cluster:  https://spruce.mongodb.com/version/6733576ecae30a00078a2988/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by blink1073** [comment]
Ah, it was just slow.  It passes if I remove pytest-timeout: https://spruce.mongodb.com/version/67356f7ff6d92b0007b61d1a/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: mongo_orchestration/servers.py [hunk 1]
```diff
@@ -418,6 +425,9 @@ def shutdown(self):
             return
         logger.info("Attempting to connect to %s", self.hostname)
         client = self.connection
+        # If we have set requireApiVersion, disable it.
+        if self.require_api_version:
+            client.admin.command("setParameter", 1, requireApiVersion=0)
```
Comment (ShaneHarvey) [inline mongo_orchestration/servers.py:430]: Is disabling requireApiVersion required before a shutdown?

Comment (blink1073) [reply mongo_orchestration/servers.py:430]: I added it based on Jesse's comment: https://jira.mongodb.org/browse/PYTHON-2520?focusedCommentId=3583776&focusedId=3583776&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3583776

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**